.devcontainer/Dockerfile

```diff
@@ -0,0 +1,59 @@
+ARG PYTHON_VERSION="3.10"
+FROM python:${PYTHON_VERSION}-slim
+
+RUN apt-get update \
+    && apt-get upgrade -y \
+    && apt-get install -y --no-install-recommends \
+        ca-certificates \
+        coreutils \
+        curl \
+        dialog \
+        dirmngr \
+        gcc \
+        git \
+        gnupg2 \
+        iproute2 \
+        libssl1.1 \
+        libxml2-dev \
+        libxslt-dev \
+        openssh-client \
+        procps \
+        psmisc \
+        rsync \
+        sudo \
+        unzip \
+        wget \
+        zip \
+        zlib1g \
+        libz-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install common python tools for development
+RUN pip install \
```

Comment:
Do we really need all of these? The minimal set of development tools in use are otherwise included in the [tests]extra and installed with `./configure --dev`

Reply:
We do if we are still relying on PIP as considering some packages will need to handle python wheel buildings.
As i heard you are working on an pip-less env, maybe this is not needed anymore